contracts/netmap/contract.go

```diff
@@ -409,11 +477,36 @@ func Netmap() []Node {
 // Current state of each node is represented in the State field. It MAY differ
 // with the state encoded into BLOB field, in this case binary encoded state
 // MUST NOT be processed.
+//
+// Deprecated: migrate to [ListCandidates].
 func NetmapCandidates() []Node {
 	ctx := storage.GetReadOnlyContext()
 	return getNetmapNodes(ctx)
 }
 
+// ListNodes provides an iterator to walk over current node set. It is similar
+// to [Netmap] method, iterator values are [Node2] structures.
+func ListNodes() iterator.Iterator {
+	return ListNodesEpoch(Epoch())
+}
+
+// ListNodesEpoch provides an iterator to walk over node set at the given epoch.
+// It's the same as [ListNodes] (and exposed as listNodes from the contract via
+// overload), but allows to query a particular epoch data if it's still stored.
+func ListNodesEpoch(epoch int) iterator.Iterator {
+	return storage.Find(storage.GetReadOnlyContext(),
+		append([]byte(node2NetmapPrefix), fourBytesBE(epoch)...),
+		storage.ValuesOnly|storage.DeserializeValues)
```

Comment:
I see that no data migration is supported, and contract storage is going to have a transition period during which the newest part of data will be stored in a new format whereas some old epoch data will be stored in old format. It's OK, but I suppose the caller's code should use `ListNodesEpoch` only after a full transition to the new storage scheme.